docs/design/eth-filter.md

```diff
@@ -0,0 +1,133 @@
+# Filter API design
+
+## Purpose
+
+The JSON-RPC Relay currently doesn't support monitoring events using filter API as is common in many other relays and is expected by many web3 tools, which makes polling for event logs, new blocks and transactions hard for developers.
+
+## Goals
+
+1. Introduce new methods:
+   1. `eth_newFilter`
+   2. `eth_uninstallFilter`
+   3. `eth_getFilterChanges`
+   4. `eth_getFilterLogs`
+
+## Non-Goals
+
+1. Decrease the load on the subscription, as this is an alternative.
+2. Increase the supported functionality of the relay.
+
+## Architecture
+
+New filter API methods can be added and handled by extending `eth` class. Adding several new methods is needed and for saving the filter IDs it needs to utilize the already available cache.
+
+### Filter Types
+
+|      Filter Type       |                                                    Description                                                     |                           Support                           |
+| :--------------------: | :----------------------------------------------------------------------------------------------------------------: | :---------------------------------------------------------: |
+|          logs          |                               Returns all logs matching filter with given filter ID.                               |                       Yes, initially                        |
```

Comment:
Can you speak to the time frame of the logs. I believe it's all the logs since the last query to the filter correct?

Reply:
Hm, we should follow `eth_getLogs` limitations. `eth_getFilterChanges` will give you all new logs since last poll and `eth_getFilterLogs` gives you all logs for the provided filters.